src/NuGetForUnity/Editor/NugetAssetPostprocessor.cs

```diff
@@ -31,11 +31,32 @@ namespace NugetForUnity
     /// </summary>
     public class NugetAssetPostprocessor : AssetPostprocessor
     {
+        private const string MaxSupportedRoslynVersion =
+#if UNITY_2022_3_OR_NEWER && !UNITY_2022_3_1 && !UNITY_2022_3_2 && !UNITY_2022_3_3 && !UNITY_2022_3_3 && !UNITY_2022_3_4 && !UNITY_2022_3_5 && !UNITY_2022_3_6 && !UNITY_2022_3_7 && !UNITY_2022_3_8 && !UNITY_2022_3_9 && !UNITY_2022_3_10 && !UNITY_2022_3_11
```

Comment:
We can't use `#if` toggles because the plugin is also released as a pre-build `.dll` so all compiler directives are already evaluated before imported into Unity.
Probably we should reuse the `UnityVersion` class from `src/NuGetForUnity/Editor/TargetFrameworkResolver.cs` witch would also make the statement a little bit easier.
Or can't we even detect the version using reflection with something like `typeof(Microsoft.CodeAnalysis.CSharp).Assembly.GetName().Version` (not testet) but as Unity has the "package" it should be possible to find the version of the assembly.

Reply:
@JoC0de I fixed `UnityVersion` to be internal instead of private and use it.

> We can't use #if toggles because the plugin is also released as a pre-build .dll so all compiler directives are already evaluated before imported into Unity.
> Probably we should reuse the UnityVersion class from src/NuGetForUnity/Editor/TargetFrameworkResolver.cs witch would also make the statement a little bit easier.

As far as I know, NugetForUnity is provided as a Unity Package with the source code, so #if seems to work too.
But as you say, it would make sense to share the UnityVersion logic. I did so.

> Or can't we even detect the version using reflection with something like typeof(Microsoft.CodeAnalysis.CSharp).Assembly.GetName().Version (not testet) but as Unity has the "package" it should be possible to find the version of the assembly.

`Microsoft.CodeAnalysis.CSharp` is not linked to the Editor or Runtime context of Unity. And it  is not present in the packages our Unity application depends on.
It is used by the Unity system to compile the application layer.
I don't know how to access this at runtime from the Editor context. Any idea?